app/src/main/java/ai/elimu/appstore/MainActivity.java

```diff
@@ -35,9 +35,18 @@ protected void onStart() {
             finish();
         } else {
             // Redirect to Activity for downloading list of Applications from REST API
-            Intent intent = new Intent(getApplicationContext(), InitialSyncActivity.class);
-            startActivity(intent);
-            finish();
+
+
+            runOnUiThread(new Runnable() {
```

Comment:
Why was `runOnUiThread` added?

Reply:
While running application on local i faced an issue where after coming back to app it was running on different thread while loading data on to Screen. So i used runOnUiThread  to in UI thread.

Reply:
> While running application on local i faced an issue where after coming back to app it was running on different thread while loading data on to Screen. So i used runOnUiThread to in UI thread.

@arvind-1 Are you saying that the application was already downloading content from the REST API when you came back to the app after having opened it previously? If that's the case, maybe it would be better if we implement code that will cancel the download if the user leaves the app (`[onStop()](https://developer.android.com/guide/components/activities/activity-lifecycle)`)?

CC @gscdev